geostore/step_function.py

```diff
@@ -180,3 +181,29 @@ def get_hash_key(dataset_id: str, dataset_version_id: str) -> str:
     return (
         f"{DATASET_ID_PREFIX}{dataset_id}{DB_KEY_SEPARATOR}{VERSION_ID_PREFIX}{dataset_version_id}"
     )
+
+
+def mark_file_as_replaced(
+    dataset_id: str,
+    current_version_id: str,
+    filename: str,
+    processing_asset_type: ProcessingAssetType,
+    processing_assets_model: Type[ProcessingAssetsModelBase],
+) -> None:
+    if current_version_id == "None":
+        return
+
+    items = processing_assets_model.query(
+        get_hash_key(dataset_id, current_version_id),
+        range_key_condition=processing_assets_model.sk.startswith(
+            f"{processing_asset_type.value}{DB_KEY_SEPARATOR}"
+        ),
+        filter_condition=(processing_assets_model.filename == filename),
+        consistent_read=True,
+    )
+
+    assert items.total_count > 1, "More than 1 file returned with the same filename"
```

Comment:
Did you mean to `assert items.total_count == 1`?

Also, the assertion message should probably include the relevant parameters to debug it, basically the dataset ID, version ID and filename.

Reply:
No, I meant to assert that total is  <2. 

It's a valid case that this query returns 0 - this is the case when this file is new in this version.